app/controllers/comments_controller.rb

```diff
@@ -17,9 +18,28 @@ def create
     end
   end
 
+  def edit; end
+
+  def update
+    if @comment.update(comment_params)
+      redirect_to @giveaway
+    else
+      render :edit
+    end
+  end
+
   private
 
   def comment_params
     params.require(:comment).permit(:commenter, :body)
   end
+
+  def set_giveaway
+    @giveaway = Giveaway.find(params[:giveaway_id])
+  end
+
+  def find_comment
+    set_giveaway
```

Comment:
I believe this is redundant here. In the before action you already set giveaway before setting the comment.

Reply:
@violeta-p Edit and Upadate actions don't work otherwise

Reply:
you add them in the array of the before_action

Reply:
Ok.